test/test_all.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2025, CARV ICS FORTH.
+ *
+ * SPDX-License-Identifier:    BSD-3-Clause
+ */
+
+/*
+ * @file		test_all.h
+ * @brief		Includes, defines and testing suite structs for test_all.c
+ */
+
+#include <stdio.h>
+#include <pthread.h>
+#include <metal/alloc.h>
+#include <metal/assert.h>
+#include <metal/atomic.h>
+#include <metal/cache.h>
+#include <metal/config.h>
+#include <metal/cpu.h>
+#include <metal/device.h>
+#include <metal/dma.h>
+#include <metal/irq.h>
+#include <metal/log.h>
+#include <metal/mutex.h>
+#include <metal/shmem.h>
+#include <metal/sleep.h>
+#include <metal/spinlock.h>
+#include <metal/sys.h>
+#include <metal/time.h>
+#include <metal/utilities.h>
+#include "metal-test.h"
+
+#include "alloc.c"
+#include "assert.c"
+#include "cache.c"
+#include "config.c"
+#include "cpu.c"
+#include "init.c"
+#include "log.c"
+#include "mutex.c"
+#include "shmem.c"
+#include "sleep.c"
+#include "spinlock.c"
+#include "utilities.c"
```

Comment:
include C file should be replaced by extern declarations. Not clean to include c file even if it is for test.
```
extern int test_alloc(void);
...
```